tests/phpunit/elementor/core/experiments/test-manager.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Elementor\Tests\Phpunit\Elementor\Core\Experiments;
+
+use Elementor\Core\experiments\Manager as Experiments_Manager;
+use Elementor\Testing\Elementor_Test_Base;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+class Test_Manager extends Elementor_Test_Base {
+
+	public function tearDown() {
+		parent::tearDown();
+
+		$experiments = $this->elementor()->experiments;
+
+		$features = $experiments->get_features();
+
+		foreach( $features as $feature_name => $feature ) {
+			$experiments->remove_feature( $feature_name );
+		}
+	}
+
+	public function test_add_feature() {
+		$experiments = $this->elementor()->experiments;
+
+		$test_feature_data = [
+			'name' => 'test_feature',
+			'unaccepted_value' => 'bla bla',
+		];
+
+		$test_set = [
+			'description' => '',
+			'status' => Experiments_Manager::STATUS_ALPHA,
+			'default' => Experiments_Manager::STATE_INACTIVE,
+			'name' => 'test_feature',
+			'state' => Experiments_Manager::STATE_DEFAULT,
+		];
+
+		$new_feature = $experiments->add_feature( $test_feature_data );
+
+		$again_added_feature = $experiments->add_feature( $test_feature_data );
+
+		$this->assertEqualSets( $test_set, $new_feature );
+
+		$this->assertEquals( null, $again_added_feature );
+	}
+
+	public function test_get_features() {
+		$experiments = $this->elementor()->experiments;
+
+		// Ensures `get_features` is called correctly even before adding any feature
+		$experiments->get_features();
```

Comment:
`add_feature` should ensure it's already initialized.

Reply:
It shouldn't, because `features` are initialized in the constructor. The problem was that before we put it in the constructor, sometimes when `get_features` had been called prior to `add_feature`, it had caused a fatal error